src/LibraryManager.Vsix/Shared/VsHelpers.cs

```diff
@@ -241,36 +241,52 @@ public static bool IsKind(this Project project, params string[] kindGuids)
             return false;
         }
 
-        public static async Task<bool> DeleteFilesFromProjectAsync (Project project, IEnumerable<string> filePaths, Action<string, LogLevel> logAction, CancellationToken cancellationToken)
+        public static async Task<bool> IsDotNetCoreWebProjectAsync(Project project)
         {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
             if (project == null || IsCapabilityMatch(project, Constants.DotNetCoreWebCapability))
```

Comment:
In the previous method this check `(project==null)` would have made sense. 
In the current however, it means that any `null` project is a `DotNetCoreWebProject`.